flight/Modules/PicoC/inc/expression.c

```diff
@@ -420,7 +420,7 @@ void ExpressionAssign(struct ParseState *Parser, struct Value *DestValue, struct
             break;
         
         case TypeArray:
-            if (SourceValue->Typ->Base == TypeArray && DestValue->Typ->FromType == DestValue->Typ->FromType && DestValue->Typ->ArraySize == 0)
+            if (SourceValue->Typ->Base == TypeArray && DestValue->Typ->ArraySize == 0)
```

Comment:
I believe this was supposed to be SourceValue->Type->FromType to confirm that the types of the variables in the array match.